addons/resource/models/resource_calendar_leaves.py

```diff
@@ -53,10 +52,14 @@ def _compute_company_id(self):
 
     @api.depends('date_from')
     def _compute_date_to(self):
-        user_tz = timezone(self.env.user.tz or self._context.get('tz') or self.company_id.resource_calendar_id.tz or 'UTC')
-        for leave in self:
-            date_to_tz = user_tz.localize(leave.date_from) + relativedelta(hour=23, minute=59, second=59)
-            leave.date_to = date_to_tz.astimezone(utc).replace(tzinfo=None)
+        """This compute serves only UX purpose within hr_holidays module (in specific view afaik).
+        When user inputs a public holiday starting day, he doesn't need to fill end date (as it is set to the end of the same day).
+        Dear Refactorer: Please remove this function from the resource module if you can (should be in hr_holidays)"""
```

Comment:
Why not replace this with a
```python
def _compute_date_to(self):
    pass  # override in other modules
```
and add move the logic to `hr_holidays/models/resource.py` yourself?